### PR TITLE
Fetch and display events

### PR DESCRIPTION
**Pull request by Omkar76**

### CODE REVIEW

File: Components/AllEventPage/EventBox.tsx [hunk 1]
```diff
@@ -1,29 +1,89 @@
...
+    fetch('https://asia-south1-nsccpccoe.cloudfunctions.net/events')
+      .then(res => {
+        if (!res.ok) {
+          throw res.statusText;
+        }
+        return res.json();
+      })
+      .then((eventResponse: EventResponse) => {
+        if (eventResponse.isError) {
+          setErrorMessage("Unknown error occured while fetching event details");
```
Comment (jhasuraj01) [inline Components/AllEventPage/EventBox.tsx:60]: API also provide `errorCode` and `errorMessage`. You can use errorCode to identify error types and you can directly display `errorMessage` to user.

Refer the below API Interfaces:

```
export interface CustomResult<T> {
    isError: false
    data: T
}
```
```
export interface CustomError {
    isError: true
    errorCode: string
    errorMessage: string
}
```

**Comment by Omkar76** [comment]
02 noting down

File: Components/AllEventPage/EventBox.tsx [hunk 2]
```diff
@@ -1,29 +1,89 @@
...
-          heading="CodeHive"
-          duration="26/01/2023, 02:00 PM - 05:00 PM IST"
-        />
-        <EventElement
-          type="Competitive Programming + Hiring Contest"
-          heading="CodeRush-X"
-          duration="29/01/2023, 09:00 PM - 12:00 AM IST"
-        />
+        {
+          (errorMessage.length != 0) ? <ErrorAlert errorMessage={errorMessage} /> : events.map(event => {
```
Comment (jhasuraj01) [inline Components/AllEventPage/EventBox.tsx:78]: Instead of checking `errorMessage.length` I guess it would be better if you check `isError`

**Review by jhasuraj01 — CHANGES_REQUESTED** [review]

File: Components/AllEventPage/EventBox.tsx (hunk 1 above)
Comment (Omkar76) [reply Components/AllEventPage/EventBox.tsx:60]: `errorCode` and `errorMessage` are missing in api response.
image

Comment (jhasuraj01) [reply Components/AllEventPage/EventBox.tsx:60]: `errorCode` and `errorMessage` will be visible only when `isError` will be `true`

Comment (jhasuraj01) [reply Components/AllEventPage/EventBox.tsx:60]: `typeof respone = CustomResult<T> | CustomError`

Comment (Omkar76) [reply Components/AllEventPage/EventBox.tsx:60]: Thanks for clarifying!

File: Components/AllEventPage/EventBox.tsx (hunk 2 above)
Comment (Omkar76) [reply Components/AllEventPage/EventBox.tsx:78]: `isError` wont available in if request fails. for example if you change endpoint from /events to /evens you get `403 forbidden` with no body. this is handled with 
```ts
  if (!res.ok) {
          throw res.statusText;
        }
        return res.json();
.
.
.catch(error => {
        setErrorMessage(error)
      }); 
```

Comment (Omkar76) [reply Components/AllEventPage/EventBox.tsx:78]: I was using on more state variable `isError` but that feels redundant. with that approach I need to set `isError` state everytime I change `errorMessage`

Comment (jhasuraj01) [reply Components/AllEventPage/EventBox.tsx:82]: okay! thanks for clarifying!!! Nice approach 👍🏻

**Comment by jhasuraj01** [comment]
@Omkar76 is your PR complete?

**Comment by Omkar76** [comment]
> @Omkar76 is your PR complete?


Done. please review.

**Review by vyankatesh-nyati — APPROVED** [review]

**Review by jhasuraj01 — APPROVED** [review]

**Pull request merged**